src/test/java/com/google/collegeplanner/SemesterSchedulerTest.java

```diff
@@ -0,0 +1,196 @@
+
+    mathClasses.add(new Section("MATH101 (MWF Late Morning)", "MATH101", null, 10, 30, null,
+        new Meeting[] {MWFLateMorning}));
+
+    chemClasses.add(new Section("CHEM101 (TuThu Late Morning)", "CHEM101", null, 10, 30, null,
+        new Meeting[] {TuThuLateMorning}));
+
+    scheduler = new SemesterScheduler(allClasses);
+
+    ArrayList<Schedule> expectedSchedules = new ArrayList<Schedule>();
+    expectedSchedules.add(new Schedule());
+
+    expectedSchedules.get(0).addClass(compSciClasses.get(0));
+    expectedSchedules.get(0).addClass(englishClasses.get(0));
+    expectedSchedules.get(0).addClass(mathClasses.get(0));
+    expectedSchedules.get(0).addClass(chemClasses.get(0));
+
+    Assert.assertEquals(expectedSchedules, scheduler.getPossibleSchedules());
+  }
+
+  @Test
+  public void multipleMeetingsPerSectionMultiplePossibleSchedules() {
+    compSciClasses.add(new Section("0CMSC101 (MWF Morning and Friday Afternoon)", "CMSC101", null,
+        10, 30, null, new Meeting[] {MWFMorning, fridayOnlyAfternoon}));
+
+    compSciClasses.add(new Section("1CMSC101 (TuThuLateAfternoon)", "CMSC101", null, 10, 30, null,
+        new Meeting[] {TuThuLateAfternoon}));
+
+    englishClasses.add(new Section("0ENGL101 (TuThu Morning, TuThuAfternoon)", "ENGL101", null, 10,
+        30, null, new Meeting[] {TuThuMorning, TuThuAfternoon}));
+
+    mathClasses.add(new Section("0MATH101 (MWF Late Morning)", "MATH101", null, 10, 30, null,
+        new Meeting[] {MWFLateMorning}));
+    mathClasses.add(new Section(
+        "1MATH101 (MWF Afternoon)", "MATH101", null, 10, 30, null, new Meeting[] {MWFAfternoon}));
+
+    chemClasses.add(new Section("0CHEM101 (TuThu Late Morning)", "CHEM101", null, 10, 30, null,
+        new Meeting[] {TuThuLateMorning}));
+
+    scheduler = new SemesterScheduler(allClasses);
+
+    ArrayList<Schedule> expectedSchedules = new ArrayList<Schedule>();
+    expectedSchedules.add(new Schedule());
+    expectedSchedules.add(new Schedule());
+    expectedSchedules.add(new Schedule());
+
+    expectedSchedules.get(0).addClass(compSciClasses.get(0));
+    expectedSchedules.get(0).addClass(englishClasses.get(0));
+    expectedSchedules.get(0).addClass(mathClasses.get(0));
+    expectedSchedules.get(0).addClass(chemClasses.get(0));
+
+    expectedSchedules.get(1).addClass(compSciClasses.get(1));
+    expectedSchedules.get(1).addClass(englishClasses.get(0));
+    expectedSchedules.get(1).addClass(mathClasses.get(1));
+    expectedSchedules.get(1).addClass(chemClasses.get(0));
+
+    expectedSchedules.get(2).addClass(compSciClasses.get(1));
+    expectedSchedules.get(2).addClass(englishClasses.get(0));
+    expectedSchedules.get(2).addClass(mathClasses.get(0));
+    expectedSchedules.get(2).addClass(chemClasses.get(0));
+
+    ArrayList<Schedule> actualSchedules = scheduler.getPossibleSchedules();
+    Assert.assertTrue(expectedSchedules.containsAll(actualSchedules) && 
```

Comment:
if you ever finding yourself using a `&&` for test assertions, you can almost always split this up to help yourself debug easier in the future. When an assertion fails, you want it to be as easy to find out what is wrong as possible.

```
Assert.assertTrue(expectedSchedules.containsAll(actualSchedules))
Assert.assertEquals(expectedSchedules.size(), actualSchedules.size());
```

In addition, while `.containsAll` will tell you that something is wrong, it won't tell you which element is missing, which is important in debugging. Here's an example where you shouldn't need to do any copying: https://github.com/googleinterns/InterviewMe/blob/e3df6cf13c3d7e256cfb56f2a4d1e251156c9972/src/test/java/com/google/sps/data/ScheduledInterviewServletTest.java#L219